sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -114,6 +114,9 @@ static BlobOutputStream pageBlobOutputStream(final PageBlobAsyncClient client, f
     @Override
     public synchronized void close() throws IOException {
         try {
+            if (this.lastError != null && this.lastError.getMessage().equals(Constants.STREAM_CLOSED)) {
+                return;
+            }
```

Comment:
Would this be better handled as an AtomicBoolean check?

```java
// Check if the stream has already been closed and mark it as closed if it hasn't.
if (!alreadyClosed.compareAndSet(false, true)) {
  // Already been closed, return early.
  return;
}
```

There may need additional checking if the last error should be rethrown instead of just a return.